src/lib/Server/Controller/Trash.php

```diff
@@ -194,7 +195,11 @@ public function restoreItem(int $trashItemId, Request $request): Values\Resource
         $trashItem = $this->trashService->loadTrashItem($trashItemId);
 
         if ($locationDestination === null) {
-            $locationDestination = $this->locationService->loadLocation($trashItem->parentLocationId);
+            try {
+                $locationDestination = $this->locationService->loadLocation($trashItem->parentLocationId);
+            } catch (NotFoundException $e) {
+                throw new ForbiddenException(/** @Ignore */ $e->getMessage());
```

Comment:
Can this exception receive `$previous`?

Reply:
`$previous` is `null` at this point 🤔

Reply:
> `$previous` is `null` at this point 🤔

What @Steveb-p meant was probably sth like this:
```suggestion
                throw new ForbiddenException(/** @Ignore */ $e->getMessage(), 1, $e);
```
This provides better debugging info stored in logs. If we catch and throw a new one, we should include a previous one. 

_Side: `1` stands for error code. By default it's `0` which traditionally means success. But that part is a personal preference ;)_

Reply:
Oh okay, done then :)